topics/icrc_34_get_delegation.md

```diff
@@ -1,50 +1,54 @@
-    * [Example RPC Request Permission](#example-rpc-request-permission)
-  * [Request](#request)
-    * [Example RPC Request](#example-rpc-request)
-    * [Response](#response)
-    * [Example RPC Response](#example-rpc-response)
+    * [Example RPC Permission Request](#example-rpc-permission-request)
+  * [icrc-25-supported-standards](#icrc-25-supported-standards)
+  * [Request params](#request-params)
+    * [Example RPC Delegation Request](#example-rpc-delegation-request)
+  * [Result](#result)
+    * [Example RPC Delegation Response](#example-rpc-delegation-response)
   * [Message Processing](#message-processing)
   * [Errors](#errors)
 <!-- TOC -->
 
 ## Summary
 
-When a relying party wants to authenticate as a user, it uses
-a [session key](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures), and
-below `icrc34_get_global_delegation` method to obtain a delegation chain that allows the session key to sign for the
-user's global identity. The obtained delegation chain **MUST** be restricted by the signer to canister targets that
-trust the relying party following the [ICRC-28](./icrc_28_trusted_origins.md) standard.
+When a relying party wants to remove wallet approval prompts and sign on behalf of its users, it uses the icrc34_delegation 
+method described below. This method generates a delegation chain, an object with a stable identifier, allowing the relying 
+party to sign transactions on behalf of the user.
 
-- Due to the canister target restriction, only the canisters that the relying party owns can be called on behalf of the user.
-- A new delegation chain must be obtained for every new canister created by the relying party.
-- The identity is the same across all relying parties, the user is not anonymous between different relying parties.
+The delegation chain's stable identifier can take two forms: an account that may include ICP, ICRC-1, and other assets 
+(Account Delegation), or an account designed exclusively for the relying party (Relying Party Delegation).
 
-## Method
+If a relying party wants to receive an Account Delegation, the `icrc34_delegation` request **MUST** include canisters it
+controls as `targets` where each canister **MUST** implement the `icrc28_trusted_origins` endpoint as per the 
+[ICRC-28](./icrc_28_trusted_origins.md) standard.
 
-**Name and Scope:** `icrc34_get_global_delegation`
+Relying Party Delegations **MUST** be restricted by the signer to individual relying parties.
 
-**Prerequisite:** Active session with granted permission scope `icrc34_get_global_delegation` or `*`.
+Signers **SHOULD** give users the choice to authenticate with their wallets (if available) or with an app-specific identifier.
 
-## Scope (according to the [ICRC-25 standard](./icrc_25_signer_interaction_standard.md))
+Relying parties **SHOULD** remove targets if they want to isolate users from the rest of the ecosystem.
```

Comment:
This could be one purpose, but shouldn't be the main purpose of Relying Party delegations. We obviously don't want to isolate users intentionally.

Also a dapp that uses Relying Party delegations could still request the list of accounts and communicate the principal from this list as user identifier across it's platform, making the user data thus non isolated from the rest of the ecosystem.

Also using a Relying Party delegation to isolate users would run into issues, since it's not an Account delegation for a principal returned in the list of accounts, it can't hold assets that can be managed by the wallet.

Instead I would recommend to use Relying Party delegations for more technical use cases, which I tried to suggest above in the summary. 

Nonetheless, above summary should attempt to clarify the difference between a Account and Relying Party delegation. This part of the spec should focus instead on clearly indicating how to get one type of delegation or the other. 

```suggestion
Relying parties **MUST** remove targets if they want to receive a Relying Party delegation.
```

Reply:
Good point.

Only thought is the difference between **must** and **should** in this context - if a relying party wants to receive an RP delegation, and they **could** receive one even if targets are included, does the clause here need to be "must"?

Reply:
I would say yes it should be `MUST` since a RP delegation is not guaranteed when targets are defined but a side effect of the user deciding to be anonymous or targets not passing ICRC-28 validation.

So to be _guaranteed_ to receive a RP delegation, a RP must remove the targets.

Reply:
Added guaranteed to the sentence since that should probably clarify it.